src/components/Users/Login.tsx

```diff
@@ -0,0 +1,126 @@
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom"; 
+
+export default function Login() {
+  const [userEmail, setUserEmail] = useState("");
+  const [userPw, setUserPw] = useState("");
+
+  const [emailValid, setEmailValid] = useState(false);
+  const [pwValid, setPwValid] = useState(false);
+
+  const [notAllow, setNotAllow] = useState(true);
+
+  const navigate = useNavigate();
+
+  const onEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUserEmail(e.target.value);
+    const regex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+    if (regex.test(userEmail)) {
+      setEmailValid(true);
+    } else {
+      setEmailValid(false);
+    }
```

Comment:
더 깔끔하게 쓰는 방법은 다음과 같습니다.

```suggestion
      setEmailValid(regex.test(userEmail));
```

Reply:
훨씬 보기 간단하네요 앞으론 조금 더 생각해보고 짜보도록 하겠습니다 !